sda/cmd/api/api.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/neicnordic/sensitive-data-archive/internal/broker"
+	"github.com/neicnordic/sensitive-data-archive/internal/config"
+	"github.com/neicnordic/sensitive-data-archive/internal/database"
+	"github.com/neicnordic/sensitive-data-archive/internal/userauth"
+	log "github.com/sirupsen/logrus"
+)
+
+var Conf *config.Config
+var err error
+var auth *userauth.ValidateFromToken
+
+func main() {
+	Conf, err = config.NewConfig("api")
+	if err != nil {
+		log.Fatal(err)
+	}
+	Conf.API.MQ, err = broker.NewMQ(Conf.Broker)
+	if err != nil {
+		log.Fatal(err)
+	}
+	Conf.API.DB, err = database.NewSDAdb(Conf.Database)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := setupJwtAuth(); err != nil {
+		log.Fatalf("error when setting up JWT auth, reason %s", err.Error())
+
+	}
+	sigc := make(chan os.Signal, 5)
+	signal.Notify(sigc, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	go func() {
+		<-sigc
+		shutdown()
+		os.Exit(0)
+	}()
+
+	srv := setup(Conf)
+	if Conf.API.ServerCert != "" && Conf.API.ServerKey != "" {
+		log.Infof("Web server is ready to receive connections at https://%s:%d", Conf.API.Host, Conf.API.Port)
```

Comment:
```suggestion
		log.Infof("Starting web server at https://%s:%d", Conf.API.Host, Conf.API.Port)
```

Less confusing if something fails at start, does not give impression that server was running, then failed.